cel/cel_test.go

```diff
@@ -1857,3 +1858,110 @@ func TestDeclareContextProto(t *testing.T) {
 		t.Fatalf("env.Compile(%s) failed: %s", expression, iss.Err())
 	}
 }
+
+type testRegexCall struct {
```

Comment:
As a usability improvement, I'd suggest exposing a new call in the `interpretable.go` similar to `NewConstValue`, but for generating a new call node called, `NewCall` or `NewInterpretableCall` which under the covers instantiates an `interpretable.evalVarArgs{}` so that clients won't have to replicate the `IsErrorOrUnknown` checks.

Reply:
That worked out to simplify things quite a bit. I've also simplified the `Factory` signature to make it easier to use as well.